internal/handlers/header_credentials_handler.go

```diff
@@ -0,0 +1,49 @@
+package handlers
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/ydataai/authentication-service/internal/configurations"
+	authErrors "github.com/ydataai/authentication-service/internal/errors"
+	"github.com/ydataai/go-core/pkg/common/logging"
+)
+
+// HeaderCredentialsHandler defines a HeaderCredentialsHandler struct.
+type HeaderCredentialsHandler struct {
+	restCtrlConfig configurations.RESTControllerConfiguration
+	logger         logging.Logger
+}
+
+// NewHeaderCredentialsHandler defines a new HeaderCredentialsHandler struct.
+func NewHeaderCredentialsHandler(logger logging.Logger,
+	restCtrlConfig configurations.RESTControllerConfiguration) CredentialsHandler {
+
+	return &HeaderCredentialsHandler{
+		restCtrlConfig: restCtrlConfig,
+		logger:         logger,
+	}
+}
+
+// Extract is an interface that extracts credential information from the header.
+func (ah *HeaderCredentialsHandler) Extract(r *http.Request) (string, error) {
+	token, err := getBearerToken(r.Header.Get(ah.restCtrlConfig.AuthHeader))
```

Comment:
this must always be authorization, it's a standard. 
Also we must check for capitalized versions, like `authorization` and `Authorization` or even `AUTHORIZATION`

Reply:
"Authorization" header must be standard, as Renato sent.
So, if the user sends it wrong, the problem is his.